src/replaceAroundStep.ts

```diff
@@ -25,6 +28,83 @@ export function suggestReplaceAroundStep(
   prevSteps: Step[],
   suggestionId: number,
 ) {
+  // This detects and handles changes from `setNodeMarkup`
+  // (https://github.com/handlewithcarecollective/prosemirror-suggest-changes/issues/7)
+  if (
+    step.insert === 1 &&
+    step.from === 0 &&
+    step.slice.size === 2 &&
+    step.gapTo === step.to - 1 &&
+    step.gapFrom === step.from + 1 &&
+    (step as any).structure === true
+  ) {
+    const { modification } = state.schema.marks;
+    if (!modification) {
+      throw new Error(
+        `Failed to apply modifications to node: schema does not contain modification mark. Did you forget to add it?`,
+      );
+    }
+
+    const newNode = step.slice.content.firstChild!;
+    const oldNode = doc.resolve(step.from).nodeAfter!;
```

Comment:
I'm _pretty_ sure that it's correct that we can assume that these are non-null, but ESLint should be complaining about this (it is for me). Can you add an eslint-disable comment for these, explaining why it's safe to assume that they're non-null?

Reply:
I couldn't come up with a way to produce null values here, but I decided to throw an error in case we encounter it instead of eslint-disable (so it will be clear if it happens to come up later)